File: src/Entity/Candidat.php
<?php

namespace App\Entity;

use App\Repository\CandidatRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Serializer\Annotation\Groups;
use Symfony\Component\Validator\Constraints as Assert;

/**
 * Candidat
 *
 * @ORM\Table(name="candidat", indexes={@ORM\Index(name="id_Offre", columns={"id_Offre"})})
 * @ORM\Entity
 */
#[ORM\Entity(repositoryClass: CandidatRepository::class)]
class Candidat
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    /**
     * @var int
     *
     * @ORM\Column(name="id_cand", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */


        /** 
        *@Groups("Candidat")
        */
    private $idCand;

    /**
     * @var string
     *
     * @ORM\Column(name="nom", type="string", length=50, nullable=false)
     */


     #[ORM\Column(length: 255)]
     /**
      * @Assert\NotBlank
      * @Assert\Length(min=4, max=15)
      *     pattern="/^[a-zA-Z0-9\s]+$/",
      *     message="Le nom du candidat ne peut contenir que des lettres, des chiffres et des espaces"
      * )
      */

      /** 
        *@Groups("Candidat")
        */


    private $nom;

    /**
     * @var string
     *
     * @ORM\Column(name="prenom", type="string", length=50, nullable=false)
     */


    #[ORM\Column(length: 255)]
     /**
      * @Assert\NotBlank
      * @Assert\Length(min=3, max=15)
      *     pattern="/^[a-zA-Z0-9\s]+$/",
      *     message="Le prenom du candidat ne peut contenir que des lettres, des chiffres et des espaces"
      * )
      */


      /** 
        *@Groups("Candidat")
        */
    private $prenom;

    /**
     * @var string
     *
     * @ORM\Column(name="cv", type="text", length=65535, nullable=false)
     */
    #[ORM\Column(length: 255)]

    /** 
        *@Groups("Candidat")
        */
        private $cv;

    /**
     * @var string
     *
     * @ORM\Column(name="email", type="string", length=50, nullable=false)
     */


     #[ORM\Column(length: 255)]
    /**
     * @Assert\NotBlank
     * @Assert\Length(min=4, max=20)
     *     pattern="/^[a-zA-Z0-9\s]+$/",
     *     message="L'email du candidat ne peut contenir que des lettres, des chiffres et des espaces"
     * )
     */

     /** 
        *@Groups("Candidat")
        */

    private $email;

    /**
     * @var int
     *
     * @ORM\Column(name="id_Offre", type="integer", nullable=false)
     */

     #[ORM\Column(length: 255)]


     /** 
        *@Groups("Candidat")
        */
    private $idService;


   
  
    public function getidCand(): ?int
    {
        return $this->idCand;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): self
    {
        $this->nom = $nom;

        return $this;
    }

    public function getPrenom(): ?string
    {
        return $this->prenom;
    }

    public function setPrenom(string $prenom): self
    {
        $this->prenom = $prenom;

        return $this;
    }
    public function getEmail(): ?string
    {
        return $this->email;
    }

    public function setEmail(string $email): self
    {
        $this->email = $email;

        return $this;
    }
    public function getCV(): ?string
    {
        return $this->cv;
    }

    public function setCV(string $cv): self
    {
        $this->cv = $cv;

        return $this;
    }
    public function getIdservice(): ?string
    {
        return $this->idService;
    }

    public function setIdservice(int $idService): self
    {
        $this->idService = $idService;

        return $this;
    }


}

File: src/desktop/entities/Candidat.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package desktop.entities;

/**
 *
 * @author USER
 */
public class Candidat {
    private int idCand;
    private String nom;
    private String prenom;
    private String email;
    private String CV;
    private int idService;

    
    public Candidat() {
    } 
    
    
    public Candidat(String nom, String prenom, String email, String CV, int idService) {
        this.nom = nom;
        this.prenom = prenom;
        this.email = email;
        this.CV = CV;
        this.idService = idService;
    }

    public Candidat(String nom, String prenom, String email) {
        this.nom = nom;
        this.prenom = prenom;
        this.email = email;
    }

   /* public Candidat(String john_Doe, String johndoeexamplecom) {
        throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
    }*/

    public Candidat(int idCand, String nom, String email, String CV) {
        this.idCand = idCand;
        this.nom = nom;
        this.email = email;
        this.CV = CV;
    }

    

    public int getIdCand() {
        return idCand;
    }

    public void setIdCand(int idCand) {
        this.idCand = idCand;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getPrenom() {
        return prenom;
    }

    public void setPrenom(String prenom) {
        this.prenom = prenom;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getCV() {
        return CV;
    }

    public void setCV(String CV) {
        this.CV = CV;
    }

    public int getIdService() {
        return idService;
    }

    public void setIdService(int idService) {
        this.idService = idService;
    }

    public Candidat(int idCand, String nom, String prenom, String email, String CV, int idService) {
        this.idCand = idCand;
        this.nom = nom;
        this.prenom = prenom;
        this.email = email;
        this.CV = CV;
        this.idService = idService;
    }

    @Override
    public String toString() {
        return "Candidat{" + "idCand=" + idCand + ", nom=" + nom + ", prenom=" + prenom + ", email=" + email + ", CV=" + CV + ", idService=" + idService + '}';
    }

    
    
    
    
}
